The following changes allow multiple instances of Emite to be created.

tguless opened the pull request:
```
@Inject
SiteChat(@LoginXmppMap  HashMap <String, LoginXmpp> loginXmppMap, LoginXmpp loginXmpp)          {           
    this.loginXmppMap = loginXmppMap;
            this.loginXmpp = loginXmpp;
            loginXmppMap.put(login, loginXmpp);


            connection = loginXmpp.xmppConnection;
    connection.setSettings(new ConnectionSettings("mainmodule/proxy", GlobalRegistry.serverInfo.chatServer));
    session = loginXmpp.xmppSession;       

    session.login(uri(login+"@" + GlobalRegistry.serverInfo.chatServer),password);

    createHablar();     
}
```

Commit:
These changes allow the user to create multiple connections to different
XMPP Servers.

ashward commented:
Thanks Ted! There's loads of changes there - a quick check looks good but I'm hoping I'll have a bit of time to go through it all more thoroughly.

Good stuff :)

tguless commented:
Small correction:

```
            @Inject
            SiteChat(@LoginXmppMap  HashMap <String, LoginXmpp> loginXmppMap, LoginXmpp loginXmpp)          {                                            
            this.loginXmppMap = loginXmppMap;
            this.loginXmpp = loginXmpp;

            loginXmpp.setInstanceId(login);

            loginXmppMap.put(login, loginXmpp);


            connection = loginXmpp.xmppConnection;
            connection.setSettings(new ConnectionSettings("mainmodule/proxy", GlobalRegistry.serverInfo.chatServer));
            session = loginXmpp.xmppSession;       

            session.login(uri(login+"@" + GlobalRegistry.serverInfo.chatServer),password);

            createHablar();                 
            }
```

Commit:
Added public MUCChatStateManager mucChatStateManager;

Commit:
This will make sure the browser AutoConfig browser module works with the
new multisession conversion.

tguless committed:
Browser Module fix, XmppSessionLogic multi instance.

tguless committed:
Confirmed that after these changes, multiple logins are working:

String loginid1 = "1";
String loginid2 = "2";

String password1 = "password1";
String password2 = "password2";

HashMap<String, LoginXmpp> loginXmppMap = ginjector.getLoginXmppMap();

LoginXmpp loginXmpp1 = ginjector.getLoginXmpp(); 
loginXmppMap.put(loginid1, loginXmpp1);
loginXmpp.setInstanceId(loginid1);         

XmppConnection connection1 = loginXmpp1.xmppConnection;
connection1.setSettings(new ConnectionSettings("mainmodule/proxy",
GlobalRegistry.serverInfo.chatServer));
XmppSession session1 = loginXmpp1.xmppSession;       
session1.login(uri(loginid1+"@" +
GlobalRegistry.serverInfo.chatServer),password1);

LoginXmpp loginXmpp2 = ginjector.getLoginXmpp(); 
loginXmppMap.put(loginid2, loginXmpp2);
loginXmpp.setInstanceId(loginid2);         

XmppConnection connection2 = loginXmpp2.xmppConnection;
connection2.setSettings(new ConnectionSettings("mainmodule/proxy",
GlobalRegistry.serverInfo.chatServer));
XmppSession session2 = loginXmpp2.xmppSession;       
session2.login(uri(loginid2+"@" +
GlobalRegistry.serverInfo.chatServer),password2);

tguless committed:
BrowserModule fixes so that it continues working with the MultiSessions
